crates/sp-domains/src/valued_trie_root.rs

```diff
@@ -245,11 +245,65 @@ where
     }
 }
 
+type MemoryDB<T> = memory_db::MemoryDB<
+    <T as TrieLayout>::Hash,
+    memory_db::HashKey<<T as TrieLayout>::Hash>,
+    DBValue,
+>;
+
+/// Gnenerate a proof-of-inclusion of the value at the given `input[index]`
+///
+/// Return `None` if the given `index` out of range or fail to generate the proof
+pub fn generate_proof<Layout: TrieLayout>(input: &[Vec<u8>], index: u32) -> Option<Vec<Vec<u8>>> {
```

Comment:
I missed this earlier. Shouldn't this return a `Storage Proof` type instead. It would be nice to have that instead of returning some bare vectors.

Reply:
`Vec<Vec<u8>>` is what returned by `trie_db::proof::generate_proof`, we can't use `StorageProof` unless they can be verified in the same way.